datastore/models.go

```diff
@@ -236,13 +236,30 @@ func (c *Client) GetModels(
 	// Switch on the datastore engines
 	if c.Engine() == MongoDB { // Get using Mongo
 		return c.getWithMongo(ctx, models, conditions, fieldResults, queryParams)
-		// todo: add page/size for mongo
 	} else if !IsSQLEngine(c.Engine()) {
 		return ErrUnsupportedEngine
 	}
 	return c.find(ctx, models, conditions, queryParams, fieldResults, timeout)
 }
 
+// GetModelCount will return a count of the model matching conditions
+func (c *Client) GetModelCount(
+	ctx context.Context,
+	model interface{},
+	conditions map[string]interface{},
+	timeout time.Duration,
+) (int64, error) {
+
+	// Switch on the datastore engines
+	if c.Engine() == MongoDB {
+		return c.countWithMongo(ctx, model, conditions)
+	} else if !IsSQLEngine(c.Engine()) {
+		return 0, ErrUnsupportedEngine
```

Comment:
Lol - we should add a todo here

What's the level of work to make this functional?

Reply:
What exactly do you want to make functional? It is just checking whether the engine is an unsupported engine (not Postgres, MySQL or SQLite)